lib/chai/core/assertions.js

```diff
@@ -319,6 +319,12 @@ function an (type, msg) {
       , 'expected #{this} to be ' + article + type
       , 'expected #{this} not to be ' + article + type
     );
+  } else if (type === 'iterable') {
```

Comment:
Another special case for the concept of "iterable" like we did with "functions". Might be good to identify other supersets like this if they already exists in the code.

Reply:
I'd like us to be cautious around adding too many super-sets and making it hard to explain how `an()` works, without extensive docs. Ideally assertions should map to observable expressions in the language, e.g. `x[Symbol.toStringTag]`, `typeof x`.